public/app/percona/pmm-dump/components/ExportDataset/ExportDataset.tsx

```diff
@@ -0,0 +1,225 @@
+/* eslint-disable @typescript-eslint/consistent-type-assertions */
+import React, { FC, useState, useEffect, useCallback, useMemo } from 'react';
+import { Field, withTypes } from 'react-final-form';
+import { useHistory } from 'react-router-dom';
+
+import { SelectableValue, DateTime, dateTime, TimeRange, AppEvents } from '@grafana/data';
+import { LinkButton, PageToolbar, DateTimePicker, useStyles2 } from '@grafana/ui';
+import appEvents from 'app/core/app_events';
+import { GrafanaRouteComponentProps } from 'app/core/navigation/types';
+import { LoaderButton } from 'app/percona/shared/components/Elements/LoaderButton';
+import { Overlay } from 'app/percona/shared/components/Elements/Overlay';
+import { MultiSelectField } from 'app/percona/shared/components/Form/MultiSelectField';
+import { useCancelToken } from 'app/percona/shared/components/hooks/cancelToken.hook';
+import { fetchActiveServiceTypesAction, fetchServicesAction } from 'app/percona/shared/core/reducers/services';
+import { getServices } from 'app/percona/shared/core/selectors';
+import { isApiCancelError } from 'app/percona/shared/helpers/api';
+import { useAppDispatch } from 'app/store/store';
+import { useSelector } from 'app/types';
+
+import { SwitchRow } from '../../../settings/components/Advanced/SwitchRow';
+import { PMMDumpService } from '../../PMMDump.service';
+
+import { GET_SERVICES_CANCEL_TOKEN, DUMP_URL } from './ExportDataset.constants';
+import { Messages } from './ExportDataset.messages';
+import { getStyles } from './ExportDataset.styles';
+import { ExportDatasetProps } from './ExportDataset.types';
+
+const { Form } = withTypes<ExportDatasetProps>();
+
+const ExportDataset: FC<GrafanaRouteComponentProps<{ type: string; id: string }>> = ({ match }) => {
+  const styles = useStyles2(getStyles);
+  const dispatch = useAppDispatch();
+  const [selectedTimerange, setSelectedTimerange] = useState<TimeRange>();
+  const { isLoading, services: fetchedServices } = useSelector(getServices);
+
+  const flattenServices = useMemo(
+    () =>
+      fetchedServices.map((data: { params: { serviceName: string } }): SelectableValue<string> => {
+        return {
+          label: data.params.serviceName,
```

Comment:
You can use an implicit return here. Also, does the data need additional type information?

FYI:
You can pass in the returned type to the map function like so:
```tsx
fetchedServices.map<SelectableValue<string>>(...)
```

Reply:
Updated.